src/EventListener/SessionExpirationListener.php

```diff
@@ -63,33 +60,37 @@ public function __construct(TokenStorageInterface $tokenStorage, ScopeMatcher $s
      */
     public function onKernelRequest(GetResponseEvent $event): void
     {
-        if (!$this->scopeMatcher->isContaoMasterRequest($event)) {
+        $request = $event->getRequest();
+        $session = $request->getSession();
+
+        if (!$event->getRequest()->hasPreviousSession()
+            || null === $session
```

Comment:
You should handle the null pointer exception like [this](https://github.com/contao/core-bundle/blob/5e73ab0b8ebaf4608fce7a72b9d6c8f3fb0b50fa/src/Security/UserChecker.php#L150-L152) unless you are explicitly supporting the "does not have a session" case.

Reply:
Well if there is no session we just don't have to check the session lifetime… no need for an exception.

Reply:
In which case is there no session?

Reply:
I have no idea why a session can be `null`, probably if session management is disabled in the application?

Reply:
Then Contao would not work anymore. As you see `null` is not an expected state.